### PR TITLE
EOS-20055: Factory cleanup support for config through mini provisione…

### PR DESCRIPTION
**Pull request by gargpalak048**
…r interface

## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Please add Problem statement here...
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Factory cleanup support for config through mini provisioner interface
  </code>
</pre>
## Solution
<pre>
  <code>
    Added Cleanup Interface Implementation
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes
    - [ ] No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Review by sandeepsgt — APPROVED** [review]
LGTM

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py [hunk 1]
```diff
@@ -463,9 +465,61 @@ def process(self):
...
+
+    def validate(self):
+        # Validate config inputs
+        from framework.utils.utility import Utility
+        Conf.load(GLOBAL_CONFIG_INDEX, global_config_path)
+        self.product = Utility.get_config_value(GLOBAL_CONFIG_INDEX,
+            "cortx>release>product")
+        if self.product is None:
+            msg = "%s - validation failure. %s" % (
+                self.name, "'Product' name is required.")
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_setup.py:485]: 'Product' name is required to restore suitable configs.

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py [hunk 2]
```diff
@@ -463,9 +465,61 @@ def process(self):
...
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def validate(self):
+        # Common validator classes to check Cortx/system wide validator
+        pass
+
+    def process(self):
+        pass
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_setup.py:515]: Add a log stating not implemented. same in restore.process()

**Review by gargpalak048 — COMMENTED** [review]
> low-level/files/opt/seagate/sspl/setup/sspl_setup.py#L469
1 blank line required before class docstring (found 0) (D203)

This is a invalid codacy analysis.
Blank Line already there #L468

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py (hunk 1 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_setup.py]: Done

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py (hunk 2 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_setup.py]: Done

**Comment by gargpalak048** [comment]
> low-level/files/opt/seagate/sspl/setup/sspl_setup.py#L502
No blank lines allowed before class docstring (found 1) (D211)

> low-level/files/opt/seagate/sspl/setup/sspl_setup.py#L518
No blank lines allowed before class docstring (found 1) (D211)

Same. No blank line is there, this is a invalid codacy comment.

**Review by satish-seagate — APPROVED** [review]

**Review by jjxsg — APPROVED** [review]
Looks good

**Comment by cdeshmukh** [comment]
test this please

**Comment by cdeshmukh** [comment]
Previous VM deployment (http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/SSPL/261/console)   failed on some salt command error, related to kafka. 

Retrigerring deployment

**Comment by cdeshmukh** [comment]
test this please

**Review by cdeshmukh — APPROVED** [review]

**Comment by cdeshmukh** [comment]
VM deployment was passed in previous run and conflicting codacy checks D203 vs D211 are marked here, no other issues seen.

**Pull request merged**